src/js/flows/executeTableSearch.js

```diff
@@ -14,16 +14,26 @@ import executeSearch from "./executeSearch"
 
 export default function executeTableSearch(
   tabId: string,
-  args: SearchArgs
+  args: SearchArgs,
+  isBlocking: boolean = false
 ): Thunk {
   return function(dispatch) {
     return new Promise((resolve, reject) => {
+      const collectedColumns = {}
+      let collectedRecords = []
       let table = brim
         .search(args.tableProgram, args.span, args.spaceId)
         .id("Table")
         .status((status) => dispatch(Viewer.setStatus(tabId, status)))
         .chan(0, (records, types) => {
           const columns = hashDescriptorKeys(types)
+
+          if (isBlocking) {
+            Object.assign(collectedColumns, columns)
+            collectedRecords = collectedRecords.concat(records)
+            return
+          }
```

Comment:
Great. I think this is the quickest way to get the feature in without changing too much code. Perfect for getting this release out. Next time we visit this, I have some ideas on ways to re-arrange it.